site/pages/dashboard.page.tsx

```diff
@@ -29,8 +30,27 @@ export interface DashboardProps {
     newDisruptionId: string;
 }
 
-const mapDisruptions = (disruptions: Disruption[]) =>
-    sortDisruptionsByStartDate(disruptions).map((disruption) => {
+const mapDisruptions = (disruptions: Disruption[]) => {
+    return sortDisruptionsByStartDate(disruptions).map((disruption) => {
+        let maxEndDate: Dayjs | null = getDate().subtract(100, "years");
```

Comment:
We dont need to add the Dayjs | null type here - it will be inferred by the IDE. 

Also, why is the maxEndDate 100 years ago?

Reply:
Removing Dayjs | null throws an error in line 50 `Type 'null' is not assignable to type 'Dayjs'`

For maxEndDate the initialisation value is a value in past so that when the iteration begins it is replaced by the value in the first validity

Reply:
I dont understand what you mean here - arent we just finding if there is a repeating validity without an end date, and if there is then we're putting in a value of a year from today?

Reply:
yes and if there is no end date then we are setting the value to null cause this null value is used in `formatDisruptionsIntoRows` function to display end date or onwards text

Reply:
I still dont understand why we need a date 100 years in the past?

Reply:
If we find the validityPeriod that we need to set a date in, cant we just insert the endDate as a year from the start date, and job done?